### PR TITLE
chore: refactor service as a prop

### PR DESCRIPTION
**Pull request by notbot00**
Closes: #83 

Refactors the redux actions, so a service can be injected instead of directly importing it.

### CODE REVIEW

**Review by dexturr — APPROVED** [review]

File: frontend/src/app-loader.tsx [hunk 1]
```diff
@@ -23,17 +19,18 @@ export function AppLoader({ children }: { children: React.ReactNode }) {
...
     state: { status, network, networkConfig },
+    actions,
     dispatch
   } = useGlobal()
 
   // Get wallets, service state and version
   useEffect(() => {
-    dispatch(initAppAction())
+    dispatch(actions.initAppAction())
   }, [dispatch])
```
Comment (mattrussell36) [inline frontend/src/app-loader.tsx:29]: chore (blocking): We should add actions to the deps array here or an eslint-ignore if we definitely dont want to. Same for actions.startServiceAction too

File: frontend/src/components/passphrase-modal/passphrase-modal.tsx [hunk 2]
```diff
@@ -28,12 +27,12 @@ interface FormFields {
...
-  const { state, dispatch } = useGlobal()
+  const { state, actions, dispatch } = useGlobal()
 
   // Register handler.open to open the passphrase modal
   useEffect(() => {
     handler.open = () => {
-      dispatch(setPassphraseModalAction(true))
+      dispatch(actions.setPassphraseModalAction(true))
     }
   }, [dispatch])
```
Comment (mattrussell36) [inline frontend/src/components/passphrase-modal/passphrase-modal.tsx:37]: chore (blocking): missing actions in deps array here

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Theres a few warnings about `actions` being missing from the deps array that we should either fix or ignore with a comment

File: frontend/src/app-loader.tsx (hunk 1 above)
Comment (notbot00) [reply frontend/src/app-loader.tsx:29]: i think i'll try to add a linter to this repo, as it's super weird to have lint-like requests on PRs by humans :D the actions object never changes after definition, so i think the effect is going to be fine without it in this case.

Comment (notbot00) [reply frontend/src/app-loader.tsx]: fixed!

File: frontend/src/components/passphrase-modal/passphrase-modal.tsx (hunk 2 above)
Comment (notbot00) [reply frontend/src/components/passphrase-modal/passphrase-modal.tsx]: fixed

**Review by dexturr — APPROVED** [review]

**Review by mattrussell36 — APPROVED** [review]

**Review by dexturr — APPROVED** [review]

**Pull request merged**